mbed_devices/_internal/resolve_target.py

```diff
@@ -0,0 +1,59 @@
+"""Resolve targets for CandidateDevice.
+
+Filesystem access is slow - this module does its best to do it lazily.
+"""
+import itertools
+import functools
+import pathlib
+from typing import Callable, Iterable, Optional
+from mbed_targets import MbedTarget, get_target_by_product_code, get_target_by_online_id
+from mbed_tools_lib.exceptions import ToolsError
+
+from mbed_devices._internal.htm_file import HTMFileContentsParser
+from mbed_devices._internal.candidate_device import CandidateDevice
+
+
+def resolve_target(candidate: CandidateDevice) -> Optional[MbedTarget]:
+    """Resolves target for given CandidateDevice if possible.
+
+    Currently, the mechanism for resolving targets relies on existence of HTM files in devices mass storage.
+    The specification of those HTM files is that they redirect to devices product page on os.mbed.com.
+    Information about Mbed Enabled requirements: https://www.mbed.com/en/about-mbed/mbed-enabled/requirements/
+    """
+    try:
+        return _determine_target_resolver(candidate.mount_points)()
+    except ToolsError:
+        return None
+
+
+def _determine_target_resolver(candidate_mount_points: Iterable[pathlib.Path]) -> Callable:
```

Comment:
Why do you use `pathlib.Path` when you could import `Path` directly? I don't mind particularly, but wondered if there was any convention around it that I was unaware of.

Reply:
I don't think there is. Looking at what I've done here it seems that I lack consistency too.
I'd like to say I tend to use fully qualified paths until it becomes annoying, but you might've noticed `mbed_targets` gets same usage. 😅